@dsplaisted @gkulin I was seeing workload manifest mismatches (between mono and emsdk) in this PR. Digging through the log files, one of Grace's manifest update tests ran just before a failure regarding a mismatched manifest (though our log files didn't capture enough info for me to confirm that it was updating the manifests of the dotnet used in the later test). Once I disabled Grace's tests, the PR passed.
https://github.com/dotnet/sdk/runs/6218772748
https://github.com/dotnet/sdk/pull/25057/commits/71683fb34f8caad59d24b6f39d46bab7d920e5f6

I'm inclined to file a bug to have Grace review those 3 tests I disabled to figure out why they are impacting other tests and merge this PR (and keep going with the helix migrations). Thoughts?

From the ubuntu log file:
```
Running /mnt/vss/_work/1/s/artifacts/bin/redist/Release/dotnet/dotnet workload install --verbosity:minimal wasm-tools
Process ID: 22477
Running /mnt/vss/_work/1/s/artifacts/bin/redist/Release/dotnet/dotnet msbuild /t:Restore /mnt/vss/_work/1/s/artifacts/tmp/Release/It_builds_wit---198ADF14/HelloWorld.csproj
Process ID: 22506
Running /mnt/vss/_work/1/s/artifacts/bin/redist/Release/dotnet/dotnet msbuild /t:Restore /mnt/vss/_work/1/s/artifacts/tmp/Release/_GivenDotnetR---C87024F7/TestAppEchoDotnetRoot.csproj
Process ID: 22533
Running /mnt/vss/_work/1/s/artifacts/bin/redist/Release/dotnet/dotnet msbuild /t:Restore /mnt/vss/_work/1/s/artifacts/tmp/Release/_GivenDotnetR---C87024F7/TestAppEchoDotnetRoot.csproj
Process ID: 22556
[31;1m    Microsoft.DotNet.Cli.Run.Tests.GivenDotnetRootEnv.ItShouldSetDotnetRootToDirectoryOfMuxer(targetFramework: "net5.0") [FAIL]
[m[37m      Expected command to pass but it did not.
[m[37m      File Name: /mnt/vss/_work/1/s/artifacts/bin/redist/Release/dotnet/dotnet
[m[37m      Arguments: msbuild /t:Restore /mnt/vss/_work/1/s/artifacts/tmp/Release/_GivenDotnetR---C87024F7/TestAppEchoDotnetRoot.csproj
[m[37m      Exit Code: 1
[m[37m      StdOut:
[m[37m      Microsoft (R) Build Engine version 17.3.0-preview-22227-03+a1fa15d13 for .NET
[m[37m      Copyright (C) Microsoft Corporation. All rights reserved.
[m[37m      
[m[37m      /mnt/vss/_work/1/s/artifacts/tmp/Release/_GivenDotnetR---C87024F7/TestAppEchoDotnetRoot.csproj : error MSB4242: SDK Resolver Failure: "The SDK resolver "Microsoft.DotNet.MSBuildWorkloadSdkResolver" failed while attempting to resolve the SDK "Microsoft.NET.Sdk". Exception: "Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestCompositionException: Workload manifest dependency 'Microsoft.NET.Workload.Emscripten' version '7.0.0-preview.3.22122.1' is lower than version '7.0.0-preview.4.22215.1' required by manifest 'microsoft.net.workload.mono.toolchain' [/mnt/vss/_work/1/s/artifacts/bin/redist/Release/dotnet/sdk-manifests/7.0.100/microsoft.net.workload.mono.toolchain/WorkloadManifest.json]
```

--- a/src/Tests/UnitTests.proj
+++ b/src/Tests/UnitTests.proj
@@ -34,13 +34,13 @@
   <Target Name="CopyHelixFiles" AfterTargets="Publish">
 
     <ItemGroup>
-      <BuiltSdks Include="$(reporoot)artifacts\bin\$(Configuration)\Sdks\**\*.*"/>
+      <BuiltSdks Include="$(RepoRoot)artifacts\bin\$(Configuration)\Sdks\**\*.*"/>
 
       <!-- All the following files are just to get full framework msbuild while avoid to duplicated arcade logic by copying
       arcade and run it -->
-      <Engfolder Include="$(reporoot)eng\**\*.*"/>
-      <DotToolsFolder Include="$(reporoot).tools\**\*.*"/>
-      <GlobalJson Include="$(reporoot)global.json"/>
+      <Engfolder Include="$(RepoRoot)eng\**\*.*"/>
+      <DotToolsFolder Include="$(RepoRoot).tools\**\*.*"/>
+      <GlobalJson Include="$(RepoRoot)global.json"/>
       <!-- Get full framework msbuild end -->
 
       <!-- include .dotnet folder. So there is no extra first run experience run during the test -->
@@ -57,13 +57,13 @@
         <DestinationFolder>eng/</DestinationFolder>
       </TestExecutionDirectoryFiles>
 
-      <FilesInHelixRoot Include="$(reporoot)artifacts\tmp\$(Configuration)\NuGet.config"/>
-      <FilesInHelixRoot Condition="$([MSBuild]::IsOSPlatform(`Windows`))" Include="$(reporoot)build\RunTestsOnHelix.cmd"/>
-      <FilesInHelixRoot Condition=" '$([MSBuild]::IsOSPlatform(`Windows`))' == 'false' " Include="$(reporoot)build\RunTestsOnHelix.sh"/>
+      <FilesInHelixRoot Include="$(RepoRoot)artifacts\tmp\$(Configuration)\NuGet.config"/>
+      <FilesInHelixRoot Condition="$([MSBuild]::IsOSPlatform(`Windows`))" Include="$(RepoRoot)build\RunTestsOnHelix.cmd"/>
+      <FilesInHelixRoot Condition=" '$([MSBuild]::IsOSPlatform(`Windows`))' == 'false' " Include="$(RepoRoot)build\RunTestsOnHelix.sh"/>
     </ItemGroup>
 
     <PropertyGroup>
-      <HelixPayloadOnHost>$(reporoot)artifacts\tmp\Helixpayload</HelixPayloadOnHost>
+      <HelixPayloadOnHost>$(RepoRoot)artifacts\tmp\Helixpayload</HelixPayloadOnHost>
     </PropertyGroup>
 
     <Copy SourceFiles="@(Engfolder)" DestinationFiles="@(Engfolder->'$(HelixPayloadOnHost)\eng\%(RecursiveDir)%(Filename)%(Extension)')" />
@@ -82,11 +82,11 @@
     <PropertyGroup>
       <HelixPreCommands Condition="!$(IsPosixShell)">call %HELIX_CORRELATION_PAYLOAD%\t\RunTestsOnHelix.cmd $(TestFullMSBuild);$(HelixPreCommands)</HelixPreCommands>
       <HelixPreCommands Condition="$(IsPosixShell)">. $HELIX_CORRELATION_PAYLOAD/t/RunTestsOnHelix.sh;$(HelixPreCommands)</HelixPreCommands>
-      <TestDotnetRoot>$(reporoot)artifacts\bin\redist\$(Configuration)\dotnet</TestDotnetRoot>
+      <TestDotnetRoot>$(RepoRoot)artifacts\bin\redist\$(Configuration)\dotnet</TestDotnetRoot>
       <TestDotnetVersion>$(Version)</TestDotnetVersion>
-      <MSBuildSdkResolverDir>$(reporoot)artifacts\bin\Microsoft.DotNet.MSBuildSdkResolver</MSBuildSdkResolverDir>
-      <HelixStage0Targz>$(reporoot)artifacts\tmp\HelixStage0.tar.gz</HelixStage0Targz>
-      <MicrosoftNETBuildExtensions>$(reporoot)artifacts\bin\$(Configuration)\Sdks\Microsoft.NET.Build.Extensions</MicrosoftNETBuildExtensions>
+      <MSBuildSdkResolverDir>$(RepoRoot)artifacts\bin\Microsoft.DotNet.MSBuildSdkResolver</MSBuildSdkResolverDir>
+      <HelixStage0Targz>$(RepoRoot)artifacts\tmp\HelixStage0.tar.gz</HelixStage0Targz>
+      <MicrosoftNETBuildExtensions>$(RepoRoot)artifacts\bin\$(Configuration)\Sdks\Microsoft.NET.Build.Extensions</MicrosoftNETBuildExtensions>
     </PropertyGroup>
 
     <TarGzFileCreateFromDirectory
@@ -126,6 +126,6 @@
   <Target Name="CreateLocalHelixTestLayout" DependsOnTargets="AppendHelixPreCommand">
     <CreateLocalHelixTestLayout
         HelixCorrelationPayload="@(HelixCorrelationPayload)"
-        TestOutputDirectory="$(reporoot)artifacts\bin\localHelixTestLayout" />
+        TestOutputDirectory="$(RepoRoot)artifacts\bin\localHelixTestLayout" />
   </Target>
 </Project>

--- a/src/Tests/dotnet-workload-install.Tests/GivenDotnetWorkloadInstall.cs
+++ b/src/Tests/dotnet-workload-install.Tests/GivenDotnetWorkloadInstall.cs
@@ -487,7 +487,7 @@ namespace Microsoft.DotNet.Cli.Workload.Install.Tests
             Directory.GetFiles(installRecordPath).Count().Should().Be(2);
         }
 
-        [Fact(Skip="Disable tests that might be updating workload manifest")]
+        [Fact(Skip="https://github.com/dotnet/sdk/issues/25175")]
         public void HideManifestUpdateCheckWhenVerbosityIsQuiet()
         {
             var command = new DotnetCommand(Log);
@@ -502,7 +502,7 @@ namespace Microsoft.DotNet.Cli.Workload.Install.Tests
         }
 
 
-        [Theory(Skip="Disable tests that might be updating workload manifest")]
+        [Theory(Skip="https://github.com/dotnet/sdk/issues/25175")]
         [InlineData("--verbosity:minimal")]
         [InlineData("--verbosity:normal")]
         public void HideManifestUpdatesWhenVerbosityIsMinimalOrNormal(string verbosityFlag)
@@ -518,7 +518,7 @@ namespace Microsoft.DotNet.Cli.Workload.Install.Tests
                 .NotHaveStdOutContaining(Workloads.Workload.Install.LocalizableStrings.AdManifestUpdated);
         }
 
-        [Theory(Skip="Disable tests that might be updating workload manifest")]
+        [Theory(Skip="https://github.com/dotnet/sdk/issues/25175")]
         [InlineData("--verbosity:detailed")]
         [InlineData("--verbosity:diagnostic")]
         public void ShowManifestUpdatesWhenVerbosityIsDetailedOrDiagnostic(string verbosityFlag)

